libclamav_rust/src/cdiff.rs

```diff
@@ -286,6 +289,32 @@ impl<'a> DelOp<'a> {
     }
 }
 
+#[derive(Debug)]
+pub struct UnlinkOp<'a> {
+    db_name: &'a str,
+}
+
+/// Method to parse the cdiff line describing an unlink operation
+impl<'a> UnlinkOp<'a> {
+    pub fn new(data: &'a [u8]) -> Result<Self, InputError> {
+        let mut iter = data.split(|b| *b == b' ');
+        let db_name = str::from_utf8(
+            iter.next()
+                .ok_or(InputError::MissingParameter("UNLINK", "db_name"))?,
+        )
+        .map_err(|_| InputError::NotUnicode("database name"))?;
+
+        if !db_name
+            .chars()
+            .all(|x: char| x.is_alphanumeric() || x == '\\' || x == '/' || x == '.')
```

Comment:
the `\\` and `/` should NOT be allowed, but the alphanumeric or `.` SHOULD be allowed. 
This is copypaste from a check elsewhere in the `cdiff.rs`. we can check if the filename is purely just `.` or `..` (that's not in the original cdiff code, but we could add it).

Reply:
Consider switching this to pathbuf checks for a parent with std::path.